File: flamesGame.js
function flamesCount(name1, name2) {
	name1 = name1.toLowerCase().split("");
	name2 = name2.toLowerCase().split("");
	var i = 0;
	var j = 0;
	while (i < name1.length) {
		while (j < name2.length) {
			if (name1[i] == name2[j]) {
				name1.splice(i, 1);
				name2.splice(j, 1);
			}
			j += 1;
		}
		i += 1;
		j = 0;
	}
	return name1.length + name2.length;
}

function flamesGame(name1, name2) {
	var number = flamesCount(name1, name2);
	var flames = ["f","l","a","m","e","s"];
	var lastIndex = 0, currIndex = 0;
	var i = flames.length;
	while (i > 1) {
		currIndex = (lastIndex + (number % i) - 1) % i;
		flames.splice(currIndex, 1);
		lastIndex = currIndex;
		i -= 1;
		// console.log(flames);
	}
	return flamesName(flames[0]);
}

function flamesName(character) {
	switch(character) {
		case "f": return "Friends";
		case "l": return "Love";
		case "a": return "Affection";
		case "m": return "Marriage";
		case "e": return "Enemies";
		case "s": return "Siblings";
	}
}

// Hard-Coded input
console.log(flamesGame("Rajan", "Suchita"));

File: flamesGame.rb
def flamesCount(name1, name2)
	name1 = name1.downcase.split("")
	name2 = name2.downcase.split("")
	i = 0
	j = 0
	while i < name1.length
		while j < name2.length
			if name1[i] == name2[j]
				name1.delete_at(i)
				name2.delete_at(j)
			end
			j += 1
		end
		i += 1
		j = 0
	end
	return name1.length + name2.length
end

def flamesGame(name1, name2)
	number = flamesCount(name1, name2)
	flames = ["f","l","a","m","e","s"]
	i = flames.length
	lastIndex = 0
	while i > 1
		currIndex = (lastIndex + (number % i) - 1) % i
		flames.delete_at(currIndex)
		lastIndex = currIndex
		i -= 1
		# print flames
		# print "\n"
	end
	print flamesName(flames[0])
end

def flamesName(character)
	case character
		when "f"
			return "Friends"
		when "l"
			return "Love"
		when "a"
			return "Affection"
		when "m"
			return "Marriage"
		when "e"
			return "Enemies"
		when "s"
			return "Siblings"
	end
end

# Hard-Coded input
# puts flamesGame("Rajan", "Anvesh")

# User input
print "Please enter first name: "
name1 = gets.chomp

print "Please enter second name: "
name2 = gets.chomp

puts flamesGame(name1, name2)
